### PR TITLE
cmake: build fixes

### PR DESCRIPTION
**Pull request by alimaredia**
This PR makes all the tests that need environment variables run in `make check` have them set automatically as long as the tests are being executed out of the proper directory.

### CODE REVIEW

**Comment by alimaredia** [comment]
@cbodley here's the PR

**Comment by trociny** [comment]
lgtm

File: src/test/cephtool-test-rados.sh [hunk 1]
```diff
@@ -14,6 +14,9 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU Library Public License for more details.
 #
+source $(dirname $0)/detect-build-env-vars.sh
+CEPH_ROOT=${CEPH_ROOT:-..}
```
Comment (cbodley) [inline src/test/cephtool-test-rados.sh]: shouldn't need to set CEPH_ROOT in scripts that run detect-build-env-vars.sh

**Comment by alimaredia** [comment]
@dzafman This is the PR I was talking about in the core meeting

**Comment by dzafman** [comment]
@alimaredia The detect-build-env-vars.sh looks reasonable.  I had a version that used $(pwd) to get an absolute path instead relative paths but I can't say that one is better than the other.

**Comment by liewegas** [comment]
@alimaredia make check test cluster tests are still failing, see jenkins output

**Comment by alimaredia** [comment]
@dachary or anyone else, since this isn't causing Jenkins problems, can this be good to merge?

**Pull request merged**